### PR TITLE
{testsdk} Move `azure-devtools`'s code to `azure-cli-testsdk`

### PR DESCRIPTION
**Pull request by jiasli**
## Background from Azure SDK team

From @lmazuel:

[`azure-devtools`](https://github.com/Azure/azure-python-devtools) used to be a repo shared across SDK and CLI to share the test infra. We forked our own version a few years ago. We did try to re-share it (see https://github.com/Azure/azure-python-devtools/pull/60 to backport SDK changes into that repo). This PR never got merged, and SDK continued to use different version of `azure_devtools`, time making them more and more different.

We made a more important refactoring, since we don’t use `vcrpy` anymore to record tests. I think we reached a level where we can now more officially say that we won’t ever merge back again those two projects. Which means to me that `azure-devtools` is now officially used only by CLI team. SDK team didn’t use this repo for that last 3 years and don’t plan to use it.

## Q&A

Q: Is `azure_devtools` is still used by Python SDKs or other tools?
A: Not used by SDK for 3 years.

Q: Where is the release pipeline for `azure_devtools`?
A: It’s used to be plugged with [TravisCI](https://github.com/Azure/azure-python-devtools/blob/master/.travis.yml).

Q: Is it a good practice to vendor/incorporate `azure_devtools` directly into `azure-cli-testsdk`?
A: I think you should vendor that code indeed. I would suggest to archive that repo and migrate the code into the CLI tests code.

## Reference

Email: _Destiny of azure_devtools_

### CODE REVIEW

File: src/azure-cli/azure/cli/command_modules/storage/tests/latest/recordings/test_storage_blob_metadata_operations.yaml [hunk 1]
```diff
@@ -94,7 +94,7 @@ interactions:
       api_version_constraint)\r\nfrom knack.util import CLIError\r\nfrom azure.cli.core.profiles
       import ResourceType\r\n\r\nfrom azure.cli.command_modules.storage._client_factory
       import MISSING_CREDENTIALS_ERROR_MESSAGE\r\nfrom ..storage_test_util import
-      StorageScenarioMixin\r\nfrom azure_devtools.scenario_tests import AllowLargeResponse\r\n\r\n\r\n@api_version_constraint(ResourceType.MGMT_STORAGE,
+      StorageScenarioMixin\r\nfrom azure.cli.testsdk.scenario_tests import AllowLargeResponse\r\n\r\n\r\n@api_version_constraint(ResourceType.MGMT_STORAGE,
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/storage/tests/latest/recordings/test_storage_blob_metadata_operations.yaml:97]: It don't think it is a good idea to upload the source code `py` for to Storage for testing. @evelyn-ys

File: src/azure-cli-testsdk/azure/cli/testsdk/scenario_tests/tests/test_utilities.py [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+                create_random_name(prefix, length=len(prefix) + i)
+            self.assertEqual(str(cm.exception), 'The randomized part of the name is shorter than 4, which may not be '
+                                                'able to offer enough randomness')
+
+    def test_get_sha1_hash(self):
+        import tempfile
+        # https://docs.python.org/3/library/tempfile.html#tempfile.NamedTemporaryFile
+        # delete=False must be set for Windows, because Windows doesn't allow opening a temporary file
+        # that is already opened.
+        with tempfile.NamedTemporaryFile(delete=False) as f:
```
Comment (jiasli) [inline src/azure-cli-testsdk/azure/cli/testsdk/scenario_tests/tests/test_utilities.py:51]: Changed the code from https://github.com/Azure/azure-python-devtools/blob/1.2.0/src/azure_devtools/scenario_tests/tests/test_utilities.py#L46-L48 to make the test pass on Windows.

File: src/azure-cli-testsdk/azure/cli/testsdk/scenario_tests/tests/test_integration_test_base.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+    def test_live_test_default_constructor(self):
+        class MockTest(LiveTest):
+            def __init__(self):
+                super(MockTest, self).__init__('sample_test')
+
+            def sample_test(self):
+                pass
+        result = MockTest().run()
+        self.assertTrue(result.skipped)
```
Comment (jiasli) [inline src/azure-cli-testsdk/azure/cli/testsdk/scenario_tests/tests/test_integration_test_base.py:67]: Changed the code from https://github.com/Azure/azure-python-devtools/blob/1.2.0/src/azure_devtools/scenario_tests/tests/test_integration_test_base.py#L67 to make the test pass on latest Python.

See https://docs.python.org/3/library/unittest.html#unittest.TestCase.run

File: src/azure-cli-testsdk/azure/cli/testsdk/scenario_tests/__init__.py [hunk 4]
```diff
@@ -3,6 +3,13 @@
 # Licensed under the MIT License. See License.txt in the project root for license information.
 # --------------------------------------------------------------------------------------------
 
+"""
+This module is vendored from
+https://github.com/Azure/azure-python-devtools/tree/1.2.0/src/azure_devtools/scenario_tests
+
+More info: https://github.com/Azure/azure-cli/pull/20601
```
Comment (jiasli) [inline src/azure-cli-testsdk/azure/cli/testsdk/scenario_tests/__init__.py:10]: Added comments for where this module come from.

**Review by evelyn-ys — APPROVED** [review]

**Review by kairu-ms — APPROVED** [review]

**Review by kairu-ms — APPROVED** [review]

**Review by zhoxing-ms — APPROVED** [review]

**Comment by jiasli** [comment]
## Required changes for extensions

If you import from `azure_devtools`, change it to `azure.cli.testsdk`, such as

```diff
- from azure_devtools.scenario_tests import AllowLargeResponse
+ from azure.cli.testsdk.scenario_tests import AllowLargeResponse
```

**Pull request merged**